dotCMS/src/main/java/com/dotmarketing/business/Role.java

```diff
@@ -163,34 +171,33 @@ public void setLocked(boolean locked) {
 		this.locked = locked;
 	}
 
-	
-	// -1  means this first
-	// 1 means Role first
-	// 0 equal
-	public int compareTo(Role o) {
-		if (!(o instanceof Role))
-			return -1;
-		Role role = (Role) o;
-		if(this.getParent().equals(this.getId())){
-			return -1;
-		}
-		if(o.getId().equals(o.getParent())){
-			return 1;
-		}
-		if(RegEX.find(role.getDBFQN(), " --> ").size() > RegEX.find(this.getDBFQN(), " --> ").size()){
-			return -1;
-		}else if(RegEX.find(role.getDBFQN(), " --> ").size() < RegEX.find(this.getDBFQN(), " --> ").size()){
-			return 1;
-		}else{
-			return 0;
+	/**
+	 * This method is intended to sort the list of roles by the length of the {@code DB_FQN} value.
+	 * This is because Roles are hierarchical, and the parents need to be imported before the
+	 * children can be imported.
+	 *
+	 * @param role The {@link Role} object to be compared.
+	 *
+	 * @return If this object goes first, returns -1. If the Role object goes first, returns 1. If
+	 * they are equal, returns 0.
+	 */
+	@Override
+	public int compareTo(final Role role) {
```

Comment:
Let's include a test that goes through the `compareTo`

Reply:
Done!